var/spack/repos/builtin/packages/scotch/package.py

```diff
@@ -4,85 +4,90 @@
 class Scotch(Package):
     """Scotch is a software package for graph and mesh/hypergraph
        partitioning, graph clustering, and sparse matrix ordering."""
+
     homepage = "http://www.labri.fr/perso/pelegrin/scotch/"
-    url      = "http://gforge.inria.fr/frs/download.php/file/34099/scotch_6.0.3.tar.gz"
+    url = "http://gforge.inria.fr/frs/download.php/latestfile/298/scotch_6.0.3.tar.gz"
```

Comment:
@tgamblin: It looks to me as though the most recent release of each version (even the release for the oldest version available) is made available at a similar `latestfile` URL.  Here are the URLs for a view versions that I sampled:
- Version 5.1.6 : http://gforge.inria.fr/frs/download.php/latestfile/298/scotch_5.1.6.tar.gz
- Version 5.1.7 : http://gforge.inria.fr/frs/download.php/latestfile/298/scotch_5.1.7.tar.gz
- Version 5.1.11 : http://gforge.inria.fr/frs/download.php/latestfile/298/scotch_5.1.11_esmumps.tar.gz